gems/smithy/lib/smithy/vise/model.rb

```diff
@@ -30,7 +30,19 @@ def operations
 
       def parse_shapes(shapes)
         shapes.each_with_object({}) do |(id, shape), h|
-          h[id] = Shape.new(id, shape)
+          h[id] =
+            case shape['type']
+            when 'service'
+              ServiceShape.new(id, shape)
+            when 'resource'
+              ResourceShape.new(id, shape)
+            when 'operation'
+              OperationShape.new(id, shape)
+            when 'structure'
+              StructureShape.new(id, shape)
+            else
+              Shape.new(id, shape)
```

Comment:
So this `Vise::Shape` will cover all the remaining types but we need to make sure we cover members for non-structure shapes (i.e. enum, intEnum, list, map and union)

Reply:
Yes, we can add some of those, as we build it out.